### ISSUE TITLE
Session Cookie: Domain and port appended

### ISSUE THREAD

**Issue by bmhm** [fisharebest/webtrees#2618]
Hi Greg,

I tried to install webtrees into a docker container for testing modules. Once it is done, I'll share the code, of course.

I found out that the cookie header "SET-COOKIE" returns this:

  `Set-Cookie: WT_SESSION=fdfcbc0fab051942b235ec1f1071acb5; path=/; domain=localhost8080; HttpOnly`

The problem is, that the login will redirect me to `localhost8080`. That means, this happens on the login page. As soon as I click "login", I'm redirected to `localhost8080` instead of `localhost:8080`.

I looked up `app/Session.php`, but I haven't figured why this happens.
I also looked up anything in the config.php or database without luck.

Any ideas what's happening here?

**Comment by bmhm** [comment]
The docker scripts are a bit messy. Is there a better way to e2e-test the `module.php`, it's activation and the tab?
https://github.com/bmhm/webtrees-jsonld

**Comment by bmhm** [comment]
Just figured out I cannot login at all using 2.0.0-b4. It just redirects me to the login page again without any error.

**Comment by fisharebest** [comment]
`localhost8080`?

This comes from the `base_url` defined in `data/config.ini.php`.

Do you set `base_url`?  What did you set it to?

**Comment by bmhm** [comment]
I did not set it to anything. I installed webtrees using the installer, and then the login did not work. Base URL is not defined in the config.ini.php.

**Comment by fisharebest** [comment]
The latest code sets the base_url during installation.

Can you try again?

**Comment by bmhm** [comment]
Sure. But will take some time. Compiling from a master.zip is impossible due to the cwd needs to be  a git repository, and cloning takes ages. I was working with a dev-dependency of 2.0.0-beta4 so far.

**Comment by fisharebest** [comment]
> and cloning takes ages

Have you tried using a shallow clone?

`git clone --depth=1`

It downloads just the latest code - without the history.  Saves ~80% of the download time/bandwidth.

**Comment by bmhm** [comment]
Thanks for the clone info.

I updated my scripts, and found out that the redirect will now work. BUT: webtrees is not installed yet. 

Is there an API to install webtrees? I think it is going to be sth like this:
```
curl --url 'http://localhost:8080/index.php?route=admin-trees-create' \
  -H 'Content-Type: application/x-www-form-urlencoded' \
  -H 'Cookie: WT_SESSION=a89783f00bb2e9250e6d7859560ebecc' \
  --data 'csrf=WUm9SKCPG6XH2ENric7ZJgefwepw5O52&tree_title=webtrees&tree_name=webtrees' 
```

Do I need to curl through each page because of the session and csrf?
But in fact, the main issue seems to be resolved.

**Comment by bmhm** [comment]
Bug: base_url gets the trailing slash appended when installing into root directory of subdomain.

eg.:
Installing to `test.domain.invalid` will result in: `base_url="http://test.domain.invalid/"`

The index will redirect to `http://test.domain.invalid///?login`.

**Comment by bmhm** [comment]
Fixed with 673fcacc80ecd2f94232e23993fd5b09eebf5120
